metrics/internal/collectors/cluster-advance-feature-use.go

```diff
@@ -208,18 +210,17 @@ func getAllStorageClasses(
 		klog.Errorf("couldn't list StorageClasses. %v", err)
 		return nil
 	}
-	if len(namespaces) == 0 {
-		return allSCs
-	}
-	var namespacedSCs []*storagev1.StorageClass
-	for _, namespace := range namespaces {
-		for _, eachSC := range allSCs {
-			if eachSC.Namespace == namespace {
-				namespacedSCs = append(namespacedSCs, eachSC)
-			}
+	// Since there can be many StorageClass not related to ceph, we filter
+	// those out based on provisioner value.
+	var odfSC []*storagev1.StorageClass
+	for _, sc := range allSCs {
+		if strings.Contains(sc.Provisioner, "rbd.csi.ceph.com") ||
```

Comment:
Instead of defining these 3 provisioners can't we just take any storage class whose provisioner contains the string "csi.ceph.com"?

Reply:
We use this selector everywhere, so I used this for consistency. Also, it's better to be specific about which provisioner we're watching for.